tests/test_env_timestamp.py

```diff
@@ -0,0 +1,75 @@
+import os
+import pytest
+import time
+import datetime
+
+
+from excavator import env_timestamp
+
+try:
+    import iso8601  # NOQA
+    iso8601_available = True
+except ImportError:
+    iso8601_available = False
+
+
+def to_timestamp(when):
+    return time.mktime(when.timetuple()) + when.microsecond / 1e6
+
+
+def assert_datetimes_almost_equal(w1, w2, delta=datetime.timedelta(microseconds=1)):
+    assert abs(w1.replace(tzinfo=None) - w2.replace(tzinfo=None)) <= delta
```

Comment:
Um, this was a work-around for rounding errors in python3 on the least signifficant (0.000001) digit in the microsecond.  This was an easy fix.  I believe it had to do with the fact that the timestamp is a float, and thus I had to deal with some imprecision when downcasting and then upcasting back to a datetime.